src/WebJobs.Extensions.DurableTask/Listener/DurableTaskScaleMonitor.cs

```diff
@@ -0,0 +1,180 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#if !FUNCTIONS_V1
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.WebSockets;
+using System.Text;
+using System.Threading.Tasks;
+using DurableTask.AzureStorage.Monitoring;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.WindowsAzure.Storage;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    internal sealed class DurableTaskScaleMonitor : IScaleMonitor<DurableTaskTriggerMetrics>
+    {
+        private readonly string functionId;
+        private readonly FunctionName functionName;
+        private readonly string hubName;
+        private readonly string storageConnectionString;
+        private readonly EndToEndTraceHelper traceHelper;
+        private readonly ScaleMonitorDescriptor scaleMonitorDescriptor;
+
+        private DisconnectedPerformanceMonitor performanceMonitor;
+
+        public DurableTaskScaleMonitor(
+            string functionId,
+            FunctionName functionName,
+            string hubName,
+            string storageConnectionString,
+            EndToEndTraceHelper traceHelper,
+            DisconnectedPerformanceMonitor performanceMonitor = null)
+        {
+            this.functionId = functionId;
+            this.functionName = functionName;
+            this.hubName = hubName;
+            this.storageConnectionString = storageConnectionString;

```

Comment:
We've already established that in certain scenarios that the connection string could be null here. If that's the case, should we go ahead and throw at this point? If I'm reading this code correctly, an app could end up into a continuous failure loop if we allow a null connection string to be set here.

Reply:
So `DurableTaskScaleMonitor` will always be created for any function app. If we were to throw here then any app that doesn't use Azure Storage -- even if they have runtime scale **disabled** -- would fail. That's why the `DisconnectedPerformanceMonitor` is only created when needed during `GetMetricsAsync` or `GetScaleStatus` -- because those functions will only be called if the user has runtime scaling enabled. I think it might be worth it to explicitly throw is the connection string is null, but it would have to be when the performance monitor is created.

Reply:
My reading of the code is that this constructor is only called by `IScaleMonitor.GetMonitor()`. Are you saying we call this interface method even if runtime scaling is disabled?

Reply:
Correct. The constructor is only called by `IScaleMonitor.GetMonitor()`, but that is called on start up [here](https://github.com/Azure/azure-webjobs-sdk/blob/7b97fd9e3b5b00ef5808eeda4439768cb947f530/src/Microsoft.Azure.WebJobs.Host/Listeners/HostListenerFactory.cs#L107). After construction, `GetMetricsAsync()` and `GetScaleStatus()` will only be called if runtime scaling is enabled [here](https://github.com/Azure/azure-functions-host/blob/3d13bf10712edceb521d6b25b7f92b1ba32a17b2/src/WebJobs.Script/Scale/FunctionsScaleMonitorService.cs#L47)

Reply:
Got it, thanks for the explanation. I think Connor and I can probably take a shot at refactoring this later as we add support for alternate backends.